File: server/src/main/java/net/cryptic_game/backend/server/client/ClientWrapper.java
package net.cryptic_game.backend.server.client;

import io.netty.channel.ChannelHandlerContext;
import net.cryptic_game.backend.data.User;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ClientWrapper {

    private static final Map<ChannelHandlerContext, Client> clients = new HashMap<>();

    public static void addClient(final ChannelHandlerContext ctx) {
        clients.put(ctx, new Client(ctx));
    }

    public static void removeClient(Client client) {
        clients.remove(client.getChannelHandlerContext());
    }

    public static Client getClient(final ChannelHandlerContext ctx) {
        return clients.get(ctx);
    }

    public static List<Client> getClients(final User user) {
        final List<Client> clients = new ArrayList<>();
        for (Client c : ClientWrapper.clients.values()) {
            if (c.getUser() != null && c.getUser().getId().equals(user.getId())) clients.add(c);
        }
        return clients;
    }

    public static int getOnlineCount() {
        return clients.size();
    }
}

File: server/src/main/kotlin/net/cryptic_game/backend/server/client/ClientWrapper.kt
package net.cryptic_game.backend.server.client

import io.netty.channel.ChannelHandlerContext
import net.cryptic_game.backend.data.user.User

object ClientWrapper {
    private val clients: MutableMap<ChannelHandlerContext, Client> = HashMap()

    @JvmStatic
    val onlineCount: Int
        get() = clients.size

    @JvmStatic
    fun addClient(ctx: ChannelHandlerContext) {
        clients[ctx] = Client(ctx)
    }

    @JvmStatic
    fun removeClient(client: Client) {
        clients.remove(client.channelHandlerContext)
    }

    @JvmStatic
    fun getClient(ctx: ChannelHandlerContext) = clients[ctx]

    @JvmStatic
    fun getClients(user: User) = clients.values.filter {
        it.user?.id == user.id
    }

}